com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/protocol/Requests.java

```diff
@@ -277,6 +281,7 @@ public static RunInTerminalRequestArguments createExternalTerminal(String[] cmds
         SETFUNCTIONBREAKPOINTS("setFunctionBreakpoints", SetFunctionBreakpointsArguments.class),
         EVALUATE("evaluate", EvaluateArguments.class),
         RUNINTERMINAL("runInTerminal", RunInTerminalRequestArguments.class),
+        REDEFINECLASSES("redefineClasses", RedefineClassesArguments.class),
```

Comment:
Yes, this will be a custom request. The design is to notify VS Code for source changes and saves whatever changes as pending at the debugger side. Then VS Code sends request to the debugger to apply code changes. 
By doing so, the hot code replace is also part of the pipeline and there will be no synchronization problems.